src/services/react-admin/patchLayerDataProvider.js

```diff
@@ -0,0 +1,61 @@
+import { GET_ONE  } from 'react-admin';
+import { RES_DATALAYER } from '../../modules/RA/ra-modules';
+
+// Patch legend from old standard to new one.
+const patchLegend = ({ stackedCircles, content, source, items = [], ...rest }) => {
+  const compatible = { ...rest };
+
+  compatible.shape = rest.shape;
+
+  if (!rest.shape) {
+    if (items.length) {
+      compatible.shape = items[0].shape || 'square';
+    }
+  }
+
+  if (stackedCircles) {
+    compatible.shape = 'stackedCircle';
+  }
+
+  if (source && !compatible.comment) {
+    compatible.comment = source;
+  }
+
+  compatible.items = items.filter(({ items: subItems }) => !subItems)
+    .map(({ radius, diameter, size, shape, ...itemRest }) => {
+      let newSize;
+      if (radius) {
+        newSize = radius; // I know, should be *2 but missused...
+      }
+      if (diameter) {
+        newSize = diameter;
+      }
+      if (size) {
+        newSize = size;
+      }
+      if (newSize) {
+        return { size: newSize, ...itemRest };
+      }
+      return itemRest;
+    });
+
+  return compatible;
+};
+
+
+const patchLayerDataProvider = nextDataProvider => async (...args) => {
+  const [type, resource, params, meta = {}] = args;
+  const { endpoint = resource } = meta;
```

Comment:
Why don't you spread all theses variables directly in the arguments part?

Reply:
Copy paste :)
Fixed.